packages/server-admin-ui/src/views/DataBrowser.js

```diff
@@ -80,20 +80,28 @@ class DataBrowser extends Component {
       
       msg.updates.forEach(update => {
         if ( update.values ) {
+          let source = update['$source']
+          if ( update.source ) {
+            if ( update.source.pgn ) {
+              source = `${source} (${update.source.pgn})`
+            } else if ( update.source.sentence ) {
+              source = `${source} (${update.source.sentence})`
+            }
+          }
           update.values.forEach(vp => {
             if ( vp.path === '' ) {
               keys(vp.value).forEach(k => {
                 context[k] = {
                   path: k,
                   value: vp.value[k],
-                  source: update['$source'],
+                  source: source,
```

Comment:
I'd rather push `pgn` and `sentence` into the object and then display them like `{data.pgn || ''}`, less nested ifs.

Or if done this way the variable should be `sourceInfo` or something. I try to stick to using the name `source` to refer to the source object and `sourceRef` to denote the $source thing.